src/msg/simple/Accepter.cc

```diff
@@ -239,21 +255,46 @@ void *Accepter::entry()
   ldout(msgr->cct,10) << "accepter starting" << dendl;
   
   int errors = 0;
+  int ch;
 
-  struct pollfd pfd;
-  pfd.fd = listen_sd;
-  pfd.events = POLLIN | POLLERR | POLLNVAL | POLLHUP;
+  struct pollfd pfd[2];
+  memset(pfd, 0, sizeof(pfd));
+  create_selfpipe(&shutdown_rd_fd, &shutdown_wr_fd);
```

Comment:
maybe this selfpipe should be created in the ctor instead of entry() so we avoid an entry() vs stop() race...